### ISSUE TITLE
google_certificate_manager_certificate: Self managed certificate update using patch instead of force replacement

### ISSUE THREAD

**Issue by jdfischer-cedreo** [hashicorp/terraform-provider-google#17312]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team)
  to expedite investigation and resolution of this issue.

### Description

When using self managed certificate in the certificate manager, update using terraform is hardly possible for now.
The provider choose to delete/create when the certificate change instead of using the [patch endpoint](https://cloud.google.com/certificate-manager/docs/reference/certificate-manager/rest/v1/projects.locations.certificates/patch). Unfortunately this will failed as the certificate is in use by an entry and the certificate map is in use by load balancer.
It would be better if changes in self_managed attributes trigger an update using a patch with updateMask _self_managed_.

The idea is to work like [gcloud command](https://cloud.google.com/certificate-manager/docs/certificates#update):
```
gcloud certificate-manager certificates update CERTIFICATE_NAME \
    --certificate-file="CERTIFICATE_FILE" \
    --private-key-file="PRIVATE_KEY_FILE" \
    --description="DESCRIPTION" \
    --update-labels="LABELS" \
    --location="REGION
```

For reference we use self managed certificate because we need our wildcard certificate in multiple project and it is the only way.

### New or Affected Resource(s)

* google_certificate_manager_certificate

### Potential Terraform Configuration

```tf 
resource "google_certificate_manager_certificate" "wildcard_certificate" {
  name        = "wildcard-example-org"
  description = "*.example.org"
  scope       = "DEFAULT"
  self_managed {
    pem_certificate = file("/etc/letsencrypt/live/example.org/fullchain.pem")
    pem_private_key = file("/etc/letsencrypt/live/example.org/privkey.pem")
  }
  project = var.project_id
}

resource "google_certificate_manager_certificate_map_entry" "wildcard_certificate_map_entry" {
  name        = "wildcard-example-org-map-entry"
  description = "Wildcard."
  map = var.certificate_map_name
  certificates = [google_certificate_manager_certificate.wildcard_certificate.id]
  hostname = "*.example.org"
  project = var.project_id
}
```

**Comment by jmvanryn** [comment]
can we get a bump up on this one.. it's been well over a year

**Comment by veronikaBu** [comment]
Yes, it would be nice to update it automatically, rather than manually. 
But since it took 9 years for GCP to update LB to use RSA 4096, it might be forever

**Comment by avocarlos** [comment]
OMG we were struggling so bad with this. Please this would be such a nice rather than the current full replacement
